src/adapter/src/coord.rs

```diff
@@ -1785,10 +1806,22 @@ impl Coordinator {
             .exactly_one()
             .expect("MV dataflow must export a sink");
         let write_frontier = self.storage_write_frontier(*sink_id);
+
         // Things go wrong if we try to create a dataflow with `as_of = []`, so avoid that.
-        if !write_frontier.is_empty() {
-            as_of.join_assign(write_frontier);
-        }
+        let as_of = if write_frontier.is_empty() {
+            min_as_of.clone()
+        } else {
+            min_as_of.join(write_frontier)
+        };
+
+        tracing::info!(
+            export_ids = %dataflow.format_export_ids(),
```

Comment:
I think it might be better to have `format_export_ids` return an `impl Display` as opposed to a `String`.

Reply:
I'm not especially worried about performance here, but your suggestion looks way cleaner! I'll make a follow-up PR.